modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -622,9 +645,8 @@ func (h Handler) prepareRequest(req *http.Request, repl *caddy.Replacer) (*http.
 	// attacks, so it is strongly recommended to only use this
 	// feature if absolutely required, if read timeouts are
 	// set, and if body size is limited
-	if (h.BufferRequests || isChunkedRequest(req)) && req.Body != nil {
-		req.Body, req.ContentLength = h.bufferedBody(req.Body)
-		req.Header.Set("Content-Length", strconv.FormatInt(req.ContentLength, 10))
+	if h.ResponseBuffers != 0 && req.Body != nil {
```

Comment:
@mholt 
I consider that this branch decides whether the proxy buffers the request body or not. Thus we should look `h.RequestBuffers` .

```suggestion
	if h.RequestBuffers != 0 && req.Body != nil {
```